gradle-jdks-settings/src/main/java/com/palantir/gradle/jdks/settings/ToolchainJdksSettingsPlugin.java

```diff
@@ -144,37 +149,44 @@ public Object invoke(Object _proxy, Method method, Object[] args) throws Throwab
                 throw e.getCause();
             }
         }
+    }
 
-        private static List<Path> getInstalledToolchains(Path gradleJdksLocalDirectory) {
-            Path installationDirectory = getToolchainInstallationDir();
-            Os os = CurrentOs.get();
-            Arch arch = CurrentArch.get();
-            try (Stream<Path> stream = Files.list(gradleJdksLocalDirectory).filter(Files::isDirectory)) {
-                return stream.map(path -> path.resolve(os.toString())
-                                .resolve(arch.toString())
-                                .resolve("local-path"))
-                        .map(path -> getToolchain(path, installationDirectory))
-                        .collect(Collectors.toList());
-            } catch (IOException e) {
-                throw new RuntimeException("Unable to list the local installation paths", e);
-            }
+    private static List<Path> getOrInstallJdkPaths(Path rootProjectDir, Path gradleJdksLocalDirectory) {
+        List<Path> jdkPaths = getConfiguredJdkPaths(gradleJdksLocalDirectory);
+        List<Path> missingJdkPaths = getMissingPaths(jdkPaths);
+        if (!missingJdkPaths.isEmpty()) {
+            logger.error(
+                    "Gradle JDK setup is enabled (palantir.jdk.setup.enabled is true) but some jdks were not"
+                            + " installed: {}. If running from Intellij, please make sure the"
+                            + " `palantir-gradle-jdks` Intellij plugin is installed"
+                            + " https://plugins.jetbrains.com/plugin/24776-palantir-gradle-jdks/versions.",
```

Comment:
I would probably add that we're installing the JDKs manually this time (so it's obvious this is not a blocker)